### PR TITLE
Test feedback refactor

### PR DESCRIPTION
**Pull request by mario-galic**
@rtyley 

Made fully concurrent, removed verbose docs, made use of existing library code.

Factored out into separate controller Tip (testing-in-production) in preparation for additional endpoint.

### CODE REVIEW

File: app/controllers/Tip.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package controllers
+
+import lib._
+import play.api.mvc._
+
+object Tip extends Controller {
```
Comment (rtyley) [inline app/controllers/Tip.scala]: Call it `TestingInProduction`!

File: app/lib/TestFeedback.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      """.stripMargin
+
+    if (testPassed(result))
+      testsPassedMsg
+    else
+      testsFailedMsg
+  }
+
+  private def testPassed(result: TestResult): Boolean =
+    result.status match {
```
Comment (rtyley) [inline app/lib/TestFeedback.scala]: `result.status == "0"`

**Comment by mario-galic** [comment]
Let me have a look at GitHub Status API before you deploy this.

**Comment by mario-galic** [comment]
Turns out its possible to use Status API in the way you intended. Travis "Triggered by Prout" build sets the status on merge_commit_sha which we can read. This makes test_feedback.sh obsolete.

File: app/lib/RepoSnapshot.scala [hunk 3]
```diff
@@ -277,4 +277,24 @@ case class RepoSnapshot(
...
+    for {
+      mergedPRs <- mergedPullRequestsFor(repo)
+      seenPRs <- Future.traverse(mergedPRs){ pr => isSeen(pr).map(bool => pr -> bool) }.map(_.collect{ case (pr, true) => pr })
+    } yield {
+      seenPRs.sorted.head // latest closed + merged + seen PR
+    }
+  }
+
+  // FIXME: How to find the latest seen checkpoint when there are multiple checkpoints?
+  def latestSeenCheckpoint = activeCheckpoints.head.name
```
Comment (mario-galic) [inline app/lib/RepoSnapshot.scala]: @rtyley how to do this using existing code?

**Comment by mario-galic** [comment]
Closed in favour of https://github.com/guardian/prout/pull/38

**Pull request closed without merging**